obj_factory.h

```diff
@@ -102,6 +114,63 @@ class ObjFactory {
         return vec[idx.val].get();
     }
 
+    bool if_exist(const std::string& uri) {
+        if (map.find(uri) == map.end()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    bool erase(const Idx<ObjType>& idx) {
+        if (idx.val >= vec.size()) {
+            return false;
+        }
+        // erase map menber
+        std::string finded_key;
+        const auto* elem_ptr = vec[idx.val].get();
+        for (const auto& elem : map) {
+            if (elem.second == elem_ptr) {
+                finded_key = elem.first;
+                break;
+            }
+        }
+        if (finded_key.empty() || map.erase(finded_key) == 0) {
+            return false;
+        }
+        // erase vec menber
+        if (shift_from(idx.val) == false) {
+            return false;
+        }
+        return true;
+    }
+
+    bool erase(const std::string& uri) {
+        if (if_exist(uri) == false) {
+            return false;
+        }
+        // erase vec menber
+        const auto* elem_ptr = map.at(uri);
+        int elem_idx = -1;
+        for (int i = 0; i < (int)vec.size(); ++i) {
```

Comment:
No need to search `uri` like that. Because uri is inside obj directly. I'll change that